### ISSUE TITLE
Specialise all

### ISSUE THREAD

**Issue by minad** [TerrorJack/inline-everything#1]
Hi @TerrorJack,

somehow related - it would be interesting to support a SPECIALISE_ALL pragma. I am using such a thing in https://github.com/minad/paripari/blob/master/README.md#basic-parser-types. 
But I am using a preprocessor instead. I didn't know about source plugins before. Are those a GHC 8.6 feature?

**Comment by TerrorJack** [comment]
@minad Yes, source plugins first landed in ghc-8.6, see [doc](https://downloads.haskell.org/~ghc/latest/docs/html/users_guide/extending_ghc.html#source-plugins) for details. The basic idea is: adding a few callbacks to `Plugin` which allow you to inspect/modify parsed/renamed/type-checked AST.

I haven't much experience with `SPECIALIZE_ALL` though. If you're considering using this plugin, you can elaborate your use case, and I may support inserting pragmas other than `INLINEABLE`, and add a little bit configuration support via command line flags and such.

**Comment by minad** [comment]
Cool, but I will wait a bit till 8.6 becomes stable with 8.6.2. Note that SPECIALISE_ALL is a custom pragma, which I defined to fix performance issues in my parser combinator lib. My preprocessor inserts SPECIALISE pragmas for every matching function.
